### ISSUE TITLE
Require a tool and recipe in settings to support XeLaTex

### ISSUE THREAD

**Issue by liqimai** [James-Yu/LaTeX-Workshop#3210]
Lots of people rely on XeLaTeX to compile tex files, especially when tex files contain language other than English, e.g. Chinese characters. I notice that there is already a recipe for LuaLaTeX in `"latex-workshop.latex.tools"`, named as "lualatexmk", but still no support for `xelatex`. 

I suggest add following settings to `latex-workshop.latex.tools` and `latex-workshop.latex.recipes` to support XeLaTeX:
```json
{
    "latex-workshop.latex.tools": [
        {
            "name": "xelatexmk",
            "command": "latexmk",
            "args": [
                "-synctex=1",
                "-interaction=nonstopmode",
                "-file-line-error",
                "-xelatex",
                "-outdir=%OUTDIR%",
                "%DOC%"
            ],
        }
    ],
    "latex-workshop.latex.recipes": [
        {
            "name": "latexmk (xelatex)",
            "tools": [
                "xelatexmk"
            ]
        }
    ],
}
```

After such modification on extension side, users can compile their source by following one-line settings:
```json
{
    "latex-workshop.latex.recipe.default": "latexmk (xelatex)",
}
``` 

### Rationale 
Without such support from extension side, there are some difficulties for users to use XeLaTex.
1. User need to come up above settings of ` "latex-workshop.latex.tools"` and `"latex-workshop.latex.recipes"` by themselves. This requires expertise in latexmk and vscode build system, which is not mastered by a certain number of latex users.
2. Even if users can come up above settings by themselves, it still has some drawbacks. Currently, vscode does not enable users to add an item to particular list in settings. Above settings will overwrite "latex-workshop.latex.tools" and "latex-workshop.latex.recipes", and thus lose all other default tools and recipes provided by extension. So, they can not be set at user level, because it will mess up projects that relies on other compilers, such as pdflatex. As a result, if let users do it, they can only set it at workspace level, and need to copy the settings every time they create an XeLaTex project. 

In summary, XeLaTex can be easily supported from extension side, but difficult from user side.

**Comment by jlelong** [comment]
I will add this new recipe.
